### ISSUE TITLE
IndexError: bpy_prop_collection[index]: index 3 out of range, size 1

### ISSUE THREAD

**Issue by PavelBlend** [rlguy/Blender-FLIP-Fluids#631]
#### System and Blend File Information

**Blender Version:** 3.6.7, blender-v3.6-release, 2023-12-11 16:28, cbd81f283d58
**Addon Version:** A FLIP Fluid Simulation Tool for Blender (v1.7.3 Release 24-NOV-2023)
**OS:** Windows-10-10.0.19045-SP0
**GPU:** AMD Radeon HD 7560D ATI Technologies Inc. 4.5.13399 Core Profile Context 15.200.1062.1004
**CPU:** AMD A8-5600K APU with Radeon(tm) HD Graphics   
**CPU Threads:** 4
**RAM:** 16 GB

**Blender Binary:** C:\progs\blender\blender-3.6.7-windows-x64\blender.exe
**Addon Path:** C:\Users\Pavel\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\flip_fluids_addon
**Renderer:** CYCLES
**Cycles Device:** CPU
**Viewport Modes:** Unknown
**Objects:** 4
**FLIP Objects:** 2
**FLIP Domains:** Cube <scene: Scene>
**Obstacle Objects:** 0
**Fluid Objects:** 1 <Export Animated: 0> <Skip Re-export: 0>
**Inflow Objects:** 0
**Outflow Objects:** 0
**Force Objects:** 0
**Cache Path:** C:\Users\Pavel\Desktop\flip\untitled1_flip_fluid_cache
**Cache Exists:** True
**Cache Logs:** 4
**Simulation Visibility:** <Viewport: Enabled> <Render: Enabled>
**Surface Visibility:** <Viewport: DISPLAY_FINAL> <Render: DISPLAY_FINAL>
**Whitewater Visibility:** N/A
**Enabled Features:** Default
**Enabled Attributes:** N/A
**Lock Interface:** Enabled
**Cycles Persistent Data:** Disabled
**Developer Tools:** Enabled
**Mixbox Plugin:** Installed
**Enabled Addons:** XRay Engine Tools, FLIP Fluids 1.7.3

#### Describe the bug

I ran the simulation and saw an error in the console:
```python
Traceback (most recent call last):
  File "C:\Users\Pavel\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\addons\flip_fluids_addon\operators\preferences_operators.py", line 1089, in get_system_info_dict
    for area in bpy.data.screens[3].areas:
IndexError: bpy_prop_collection[index]: index 3 out of range, size 1
```

#### To Reproduce

I don't know when this error appeared. I accidentally saw it in the console when I was reading baking statistics.

Update:
I think I understand when the error occurs. It appears when I save a blend file (ctrl+S).

**Comment by rlguy** [comment]
Thanks for the report!

This error shouldn't cause any problems, but we can avoid this error in a future update (FLIP Fluids 1.7.4 or later, release TBA). The cause of this error is that the addon is trying to find information about the viewport shading mode, but this information is not always available. This can be fixed by handling the case where the viewport screen is not available.

**Comment by rlguy** [comment]
This has been fixed in FLIP Fluids version 1.7.4 which is now available.
